stout/flags.cc

```diff
@@ -72,6 +127,40 @@ void Parser::AddOrExit(
 
 ////////////////////////////////////////////////////////////////////////
 
+bool Parser::SetCurrentMessageForSubcommand(
+    const std::string& arg,
+    google::protobuf::Message* message) {
+  const google::protobuf::Descriptor* descriptor = message->GetDescriptor();
+
+  for (int i = 0; i < descriptor->field_count(); ++i) {
+    const google::protobuf::FieldDescriptor* field = descriptor->field(i);
+    if (field->message_type() == nullptr) {
+      continue;
+    } else {
+      if (const google::protobuf::OneofDescriptor* real_oneof =
+              field->real_containing_oneof();
+          real_oneof == nullptr) {
+        continue;
+      } else {
+        if (field->name() == arg) {
+          cur_subcommand_message_ =
+              message->GetReflection()->MutableMessage(message, field);
+          return true;
+        } else {
+          if (SetCurrentMessageForSubcommand(
```

Comment:
I'm not understanding this recursive call ... shouldn't you only be able to find the subcommand in the current level?

Seems like we need to keep around the last subcommand that we saw, and then only check if that subcommand's message has a `oneof subcommand` field that matches, and if not, then assume the command line argument was not a subcommand. And of course, the first time around we won't have a previous subcommand message, it'll just be the top-level message.